fix-for-sles12-disable-registry.patch

```diff
@@ -0,0 +1,10 @@
+--- usr/sbin/registercloudguest.orig
++++ usr/sbin/registercloudguest
+@@ -185,6 +185,7 @@ def setup_registry(registration_target, clean='registry'):
+     clean == all -> cleans repository and registry setup
+     clean == registry -> clean only registry artifacts
+     """
++    return None
+     user, password = utils.get_credentials(
+         utils.get_credentials_file(registration_target)
+     )
```

Comment:
This patch should also contain
```
def is_registry_registered(registry_server_name):
    return True
```

https://github.com/SUSE-Enceladus/cloud-regionsrv-client/pull/183/files#diff-84460adc47c95f16ae220daac1882a7bc19b595d1ba01650e286b666ef14f2a4R1671

Reply:
And we need to patch out the hosts entry in SLE 12, i.e.
https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/lib/cloudregister/registerutils.py#L75 needs to be skipped

Reply:
Hmm, are you sure. If we disable everything related to the registry setup (which is what the patch does) we will never reach the mentioned code paths

Reply:
@schaefi yes. The `add_hosts_entry` function is called to make `smt-$SCP.susecloud.net` resolvable and with the addition of the registry it now also makes the registry resolvable. But on SLE 12 we do not support containers so we do not want a registry entry in the `/etc/hosts/file. The SMT object is created with information from the region server and the region server will send the registry information since it does not differentiate between SLE12 or SLE15 systems and as such `if smt_server.get_registry_FQDN()` will also be true on SLE 12 clients.

Reply:
Hrm, yes I see. There is this code 

```
if smt_server.get_registry_FQDN():
        entry += '%s\t%s\n' % (
            smt_ip,
            smt_server.get_registry_FQDN()
        )
```

in `add_hosts_entry()` and it is called unconditionally, thus also for SLE12

ok yes this needs patching or a proper refactoring. imho it would be nice to have a generic add_hosts_entry method and two specific methods that says `add_rmt_hosts_entry` and `add_registry_hosts_entry` 

But I will not do this refactoring now and just adapt the SLE12 patch... we don't have time for this now

Reply:
done